src/lib/integration_test_cloud_engine/stack_driver_log_engine.ml

```diff
@@ -275,24 +269,40 @@ let parse_event_from_log_entry ~logger ~network log_entry =
   in
   let%bind payload = find json log_entry [ "jsonPayload" ] in
   let%map event =
-    if
-      Result.ok (find bool payload [ "puppeteer_script_event" ])
-      |> Option.value ~default:false
-    then
-      let%map msg =
-        parse (parser_from_of_yojson Puppeteer_message.of_yojson) payload
-      in
-      match msg.event_type with
-      | "node_offline" ->
-          [%log spam] "hitting node_offline event from puppeteer" ;
-          Event_type.Event (Event_type.Node_offline, ())
-      | _ ->
-          failwith "Could not process a puppeteer message from the logs"
-    else
-      let%bind msg =
-        parse (parser_from_of_yojson Logger.Message.of_yojson) payload
-      in
-      Event_type.parse_event msg
+    let%bind msg
+        (* : ([ `Daemon_origin of Logger.Message.t | `Puppeteer_origin of Puppeteer_message.t ]) *)
+        =
+      if
+        Result.ok (find bool payload [ "puppeteer_script_event" ])
+        |> Option.value ~default:false
+      then
+        let%map res =
+          parse (parser_from_of_yojson Puppeteer_message.of_yojson) payload
+        in
+        `Puppeteer_origin res
+      else
+        let%map res =
+          parse (parser_from_of_yojson Logger.Message.of_yojson) payload
+        in
+        `Daemon_origin res
+    in
+    [%log spam] "parsing event" ;
+    Event_type.parse_event msg
```

Comment:
@nholland94 

I was just playing around with different ways of writing the parse_event function.  we could do it with separate `Event_type.parse_daemon_log` and `Event_type.parse_puppeteer_log`, or we could use that anonymous ADT [ `Daemon_origin of Logger.Message.t | `Puppeteer_origin of Puppeteer_message.t ] to unify the input type.  up to whatever you feel is cleaner

Reply:
Instead of passing an ADT input to `parse_event`, I think it makes more sense now to just have 3 different interfaces for parsing events: `parse_error_log`, `parse_daemon_event`, `parse_puppeteer_event`. Then you won't need an ADT here, you can just inline calls to `parse_*` in the branches of this function.

Reply:
wouldn't it just be `parse_daemon_event`, and `parse_puppeteer_event`, why would we have a parse_error_log? the error case just happens if somehow the other ones fail, no?